backends/s3/credentials.go

```diff
@@ -0,0 +1,53 @@
+package s3
+
+import (
+	"os"
+
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// K8sSecretProvider is an implementation of Minio's credentials.Provider,
+// allowing to read credentials from Kubernetes secrets, as described in
+// https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure.
+type K8sSecretProvider struct {
+	// Path to the fiel containing the access key,
+	// e.g. /etc/s3-secrets/access-key.
+	AccessKeyFilename string `json:"access_key_file"`
```

Comment:
JSON name does not match the variable name. In this case it would commonly be called `access_key_filename`, or the variable named `AccessKeyFile`. There is no good reason to deviate from this pattern here.